geopandas/geodataframe.py

```diff
@@ -232,8 +232,15 @@ def to_json(self, na='null', show_bbox=False, **kwargs):
 
         The remaining *kwargs* are passed to json.dumps().
 
+        Noite that if the CRS is set on the geometry it's exported as epsg:4326
+        to meet the 2016 geojson specification.
+
         """
-        return json.dumps(self._to_geo(na=na, show_bbox=show_bbox), **kwargs)
+        if self.crs:
+            df = self.to_crs(epsg=4326)
+        else:
+            df = self
+        return json.dumps(df._to_geo(na=na, show_bbox=show_bbox), **kwargs)
```

Comment:
Can you add a test for CRS in [test_to_json()](https://github.com/geopandas/geopandas/blob/46e50fe5a772944b325bc3c8806f4f96da76a0d8/geopandas/tests/test_geodataframe.py#L208-L212)?